Create IntakeSubsystem.java

SMKettle opened the pull request:
Kirby

SMKettle committed:
Create IntakeSubsystem.java

kirbt commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,22 @@
+	public static final double INTAKE_FAST_OUT_SPEED = -0.7;
+
+	IntakeSubsystem() {
+		CANSparkMax INTAKE_MOTOR_BACK = new CANSparkMax(INTAKE_MOTOR_BACK_ID, MotorType.kBrushless);
```
I don't think you included all of the files you've changed/modified in your pull request

jamesdooley4 replied:
Specifically, `INTAKE_MOTOR_BACK_ID` and the other motor constants aren't defined anywhere

SMKettle committed:
updated Intake Subsystem and hardware brilliant

because i forgot

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,64 @@
+        motor.set(speed);
+    }
+
+    public void stopMotor(CANSparkMax motor){
```
This shouldn't be public, as nothing outside the class should use it.

jamesdooley4 replied:
Or have it call `.stop()` on all the motors in the class.

SMKettle replied:
So if I do 
`void stopMotor() {                                                                                              
intakeMotor1.set(0); //sets speed to 0                                                                  
}`
then that should work right?

jamesdooley4 replied:
Yep, that's correct.

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,64 @@
+
+    }
+    
+    public void intake(double speed, CANSparkMax motor){
+        motor.set(speed);
```
This needs to use one of the class member variables, as nothing outside this class should have access to have a CANSparkMax instance.

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,64 @@
+
+    public IntakeSubsystem() {
+
+        intakeMotorBack = new CANSparkMax(intakeMotorBackID, MotorType.kBrushless);
+        intakeMotorRight = new CANSparkMax(intakeMotorRightID, MotorType.kBrushless);
+        intakeMotorFront = new CANSparkMax(intakeMotorFrontID, MotorType.kBrushless);
+        intakeMotorLeft = new CANSparkMax(intakeMotorLeftID, MotorType.kBrushless);
+
+        conveyorBeltOne = new CANSparkMax(conveyorBeltOneID, MotorType.kBrushless);
+        conveyorBeltTwo = new CANSparkMax(conveyorBeltTwoID, MotorType.kBrushless);
```
Need to also configure these motors, preferably by writing and calling method like `resetMotors()` at https://github.com/robototes/ChargedUp2023/blob/f25191718390d11df9ddda789a0f0d5c35686734/src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java#L124

Yumnah2 replied:
Completed 👍

TAKBS2412 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,64 @@
+
+    
+    
+    /*
+    public void takeIn(double speed, CANSparkMax motor, CANSparkMax motor2){
+        motor.set(speed);
+        motor2.set(speed); 
+    }
+
+    public void takeIn(double speed, CANSparkMax motor, CANSparkMax motor2, CANSparkMax motor3){
+        motor.set(speed);
+        motor2.set(speed);
+        motor3.set(speed);
+    }*/
```
Delete commented out code (and run spotless to fix formatting).

TAKBS2412 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,64 @@
+	public static final double INTAKE_IN_SPEED = 0.3;
+	public static final double INTAKE_FAST_OUT_SPEED = -0.7;
+
+    CANSparkMax intakeMotorBack;
+    CANSparkMax intakeMotorRight;
+    CANSparkMax intakeMotorFront;
+    CANSparkMax intakeMotorLeft;
+
+    CANSparkMax conveyorBeltOne;
+    CANSparkMax conveyorBeltTwo;
```
Nit: should be `private final` (should not be used outside this class and should not be modified).

SMKettle replied:
Ight.

SMKettle committed:
New IntakeSubsystem based on changes

gulp

SMKettle committed:
ACTUAL REAL COMMIT

Don't mind the old commit that I made a few minutes ago

jamesdooley4 commented on src/main/java/frc/team2412/robot/Hardware.java:
```diff
@@ -11,6 +11,22 @@ public class Hardware {
+	public static final int INJEST_2 = 45;
+	public static final int INJEST_3 = 46;
+
+	public static final int INDEX_1 = 47;
+	public static final int INDEX_2 = 48;
+	public static final int INDEX_3 = 49;
+
+	public static final int FEEDER_1 = 50;
+	public static final int FEEDER_2 = 51;
+	public static final int FEEDER_3 = 52;
+	public static final int FEEDER_4 = 53;
```
Recommend not adding these until they are used, and there are good names for them.

Yumnah2 replied:
We checked in with mech to know how many motors they're using and we renamed them with directions

jamesdooley4 commented on src/main/java/frc/team2412/robot/Hardware.java:
```diff
@@ -11,6 +11,22 @@ public class Hardware {
 
-	// intake [40 - 49]
+	// intake [40 - 60]
+	public static final int INTAKE_1 = 40;
+	public static final int INTAKE_2 = 41;
+	public static final int INTAKE_3 = 42;
+	public static final int INTAKE_4 = 43;
 
+	public static final int INJEST_1 = 44;
+	public static final int INJEST_2 = 45;
+	public static final int INJEST_3 = 46;
```
Please talk with mech to come up with reasonable names for these that are immediately obvious to programmers or mech folks working on the robot, so it's clear where to look when someone says "check if intake_\<name\> is running the correct direction"

jamesdooley4 replied:
Ditto for the injest motor ID names

jamesdooley4 replied:
And the `CANSparkFlex` variable names below

Yumnah2 replied:
They did not specify wanting specific names so we named them based on directions such as intakeMotorFront, back, etc

jamesdooley4 replied:
Using directions sounds great.

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,104 @@
+	public static final double INGEST_SPEED = 0.2;
+	public static final double INDEX_SPEEED = 0.2;
+	public static final double FEEDER = 0.25; // YUMNAH please initialize THIS
+	public static final double DEEPS = 2.0;
```
Delete, unused

Yumnah2 replied:
done as well

jamesdooley4 replied:
Still seeing `DEEPS`

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,104 @@
+
+		injestMotor1 = new CANSparkFlex(INJEST_1, MotorType.kBrushless);
+		injestMotor2 = new CANSparkFlex(INJEST_2, MotorType.kBrushless);
+		injestMotor3 = new CANSparkFlex(INJEST_3, MotorType.kBrushless);
```
This should crash when the robot starts. Please run this in the simulator as part of your pre-PR review.

SMKettle committed:
commit

acknowledged the suggestions stated by Mr. Dooley and followed suit with them

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -1,64 +1,112 @@
+		injestMotorRight.burnFlash();
+	}
+
+	private void intake() {
+		intakeMotor(intakeMotorFront);
+		intakeMotor(intakeMotorBack);
+		intakeMotor(intakeMotorLeft);
+		intakeMotor(intakeMotorRight);
+	}
+
+	public void intakeMotor(CANSparkFlex motor) {
```
Swap the `public` & `private` for these. `intake()` should be public (callable by commands) and `intakeMotor` private (only used by this class)

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -1,64 +1,112 @@
+	}
+
+	public void resetMotors() {
+		intakeMotorFront.restoreFactoryDefaults();
+		intakeMotorBack.restoreFactoryDefaults();
+		intakeMotorLeft.restoreFactoryDefaults();
+		intakeMotorRight.restoreFactoryDefaults();
+
+		injestMotorLeft.restoreFactoryDefaults();
+		injestMotorRight.restoreFactoryDefaults();
+
+		intakeMotorFront.setIdleMode(IdleMode.kBrake);
+		intakeMotorBack.setIdleMode(IdleMode.kBrake);
+		intakeMotorLeft.setIdleMode(IdleMode.kBrake);
+		intakeMotorRight.setIdleMode(IdleMode.kBrake);
+
+		injestMotorLeft.setIdleMode(IdleMode.kBrake);
+		injestMotorRight.setIdleMode(IdleMode.kBrake);
+
+		intakeMotorFront.setInverted(true);
+		intakeMotorBack.setInverted(true);
+		intakeMotorLeft.setInverted(true);
+		intakeMotorRight.setInverted(true);
+
+		injestMotorLeft.setInverted(true);
+		injestMotorRight.setInverted(true);
```
To avoid this super-long function, it'd be better to create a private helper function (maybe `configureMotor(CANSparkFlex motor)`) that configures that one motor, and just have `resetMotors()` be a bunch of calls like this:
```
configureMotor(intakeMotorFront);
configureMotor(intakeMotorBack);
...
```

As long as the calls are in the same order as the class variable definitions above, it's easy to look at them to make sure all the motors are configured.  And in the future, the `configureMotor(..)` helper function can be updated to take parameters for any settings that end up being set to different values (such as current limit, inverted or not, etc.)

Yumnah2 replied:
All done, thanks for the suggestion

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -1,64 +1,112 @@
-    }*/
-
-
+	private void stopIntakeMotors() {
```
Recommend making this `public` so there's a way for commands to stop the intake motors. That's often useful to conserve power so the other subsystems have more current availble.

Yumnah2 committed:
Streamlining configuration, adding motor methods, starting shuffleboard

Addressing commit issues, changing structure of motor configuration, starting shuffleboard, and adding motor methods

Yumnah2 committed:
spotless :)

running spotless please work

TAKBS2412 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,129 @@
+		motor.burnFlash();
+	}
+
+	public void resetMotors() {
```
We should call `resetMotors()` somewhere.

Yumnah2 replied:
fixed, thank you

TAKBS2412 reviewed (changes requested):
Are we planning to add sensor logic to this PR or will that be done separately?

SMKettle committed:
Got rid of unnecessary stuff

SMKettle committed:
Merge branch 'main' of https://github.com/SMKettle/Crescendo2024intake

SMKettle committed:
spotless apply!

Yumnah2 committed:
fixing github goofs

Adding changes made that didnt commit properly

SMKettle committed:
Create SetSpeedCommand.java

kirbt commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,127 @@
+
+	// intake motor methods
+	private void intake(CANSparkFlex motor) {
+		motor.set(intakeSpeedEntry.getDouble(INTAKE_IN_SPEED));
```
The motor speed entries in these methods don't seem to be declared nor initialized anywhere in the code

SMKettle committed:
goated commit

SMKettle committed:
goated commit 2

SMKettle committed:
goated commit 3

SMKettle committed:
i delet3d it

SMKettle committed:
spotless

Description

jamesdooley4 commented on src/main/java/frc/team2412/robot/Hardware.java:
```diff
@@ -9,6 +9,18 @@ public class Hardware {
 
 	// climb [20 - 29]
 
+	// shooter [30 - 39]
+
```
Delete, and move the below intake section after the launcher section below (to keep the numbers in order)

jamesdooley4 commented on src/main/java/frc/team2412/robot/commands/intake/IndexStopCommand.java:
```diff
@@ -0,0 +1,18 @@
+
+	@Override
+	public void initialize() {
+		intakeSubsystem.indexOut();
+	}
```
Should be indexStop() ?

TAKBS2412 replied:
? curious why this was resolved.

eddiert1239 replied:
my fault, i thought i was going to finish addressing changes at the end of last meeting

jamesdooley4 commented on src/main/java/frc/team2412/robot/commands/intake/IntakeOutCommand.java:
```diff
@@ -0,0 +1 @@
+
```
Missing code?

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,125 @@
+	public static final double FEEDER_IN_SPEED = 0.3;
+	public static final double FEEDER_OUT_SPEED = -0.3;
+
+	public static final double SPEAKER_SHOOT_SPEED = 0.5;
+
```
Would delete this for now, as it's not yet used.

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,125 @@
+	public static final double SPEAKER_SHOOT_SPEED = 0.5;
+
+	// Motors
+	private final CANSparkFlex intakeMotorFront;
+	private final CANSparkFlex intakeMotorBack;
+	private final CANSparkFlex intakeMotorLeft;
+	private final CANSparkFlex intakeMotorRight;
```
Intake motors are all Neo 550 motors, which use [CANSparkMax](https://codedocs.revrobotics.com/java/com/revrobotics/cansparkmax) instead of CANSparkFlex

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,125 @@
+	}
+
+	public void intakeOut() {
+		intakeMotorFront.set(INTAKE_IN_SPEED);
```
INTAKE_OUT_SPEED

TAKBS2412 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,125 @@
+		resetMotors();
+	}
+
+	public void configureMotor(CANSparkFlex motor) {
```
This and `resetMotors()` should be `private`, as we should only call them inside the `IntakeSubsystem`.

eddiert1239 committed:
Update IndexStopCommand.java

eddiert1239 committed:
Addressed changes

eddiert1239 committed:
created intake out command

oops

eddiert1239 committed:
imported cansparkmax

eddiert1239 committed:
spotless and hardware reorg

put launcher before intake because their motors are before ours

TAKBS2412 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,146 @@
+		resetMotors();
+	}
+
+	// commented out since we have two motor types now
+	// private void configureMotor(CANSparkFlex motor) {
+	//	motor.restoreFactoryDefaults();
+	//	motor.setIdleMode(IdleMode.kBrake);
+	//	motor.setSmartCurrentLimit(20);
+	//	motor.burnFlash();
+	// }
```
For this we can use [`CANSparkBase`](https://codedocs.revrobotics.com/java/com/revrobotics/cansparkbase) which is the superclass of both `CANSparkFlex` and `CANSparkMax` and pass in the motors as before.

eddiert1239 replied:
that's so sick

eddiert1239 committed:
used cansparkbase to use configure motors again

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,125 @@
+	// Shuffleboard
+	private final GenericEntry setIntakeSpeedEntry =
+			Shuffleboard.getTab("Intake")
+					.addPersistent("Intake speed - ", INTAKE_IN_SPEED)
```
Should be "Intake in speed", as that's what being set.

jamesdooley4 replied:
Best to also rename the variable names to add `In` as well. For this that'd be setIntakeInSpeedEntry

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,125 @@
+					.getEntry();
+
+	private final GenericEntry setIndexSpeedEntry =
+			Shuffleboard.getTab("Intake").add("Index speed - ", INDEX_IN_SPEED).withSize(1, 1).getEntry();
```
Should be "Index in speed", as that's what being set.

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,125 @@
+
+	private final GenericEntry setFeederSpeedEntry =
+			Shuffleboard.getTab("Intake")
+					.add("Feeder Speed - ", FEEDER_IN_SPEED)
```
Should be "Feeeder in speed", as that's what being set.

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,125 @@
+
+	// intake methods
+	public void intakeIn() {
+		intakeMotorFront.set(INTAKE_IN_SPEED);
```
Should use setIntakeInSpeedEntry .getDouble(INTAKE_IN_SPEED);

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,125 @@
+
+	// index methods
+	public void indexIn() {
+		indexMotor.set(INDEX_IN_SPEED);
+	}
```
setIndexInSpeedEntry.getDouble(INDEX_IN_SPEED)

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,125 @@
+
+	// feeder methods
+	public void feederIn() {
+		feederMotor.set(FEEDER_IN_SPEED);
+	}
```
Use setFeederInSpeedEntry

eddiert1239 committed:
actually used shuffleboard

eddiert1239 committed:
spotless

eddiert1239 committed:
actually actually used shuffleboard
